### PR TITLE
Doc: Update COPY FROM options

### PR DESCRIPTION
**Pull request by msudberg**
Add HEADER to COPY FROM options for FORMAT CSV.

### Motivation
 
  * This PR fixes a recognized bug. #9072 
  

- [x This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR adds a release note for any
  [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

File: doc/user/content/sql/copy-from.md [hunk 1]
```diff
@@ -24,8 +24,9 @@ Name | Permitted values| Default value | Description
...
-`ESCAPE` | Single-quoted strings | `QUOTE`'s value | Specifies the character to allow instances of the `QUOTE` character to be parsed literally as part of a column's value. _`FORMAT CSV` only_
+`QUOTE` | Single-quoted one-byte character | `"` | Specifies the character to signal a quoted string, which may contain the `DELIMITER` value (without beginning new columns). To include the `QUOTE` character itself in column, wrap the column's value in the `QUOTE` character and prefix all instance of the value you want to literally interpret with the `ESCAPE` value. _`FORMAT CSV` only._
+`ESCAPE` | Single-quoted strings | `QUOTE`'s value | Specifies the character to allow instances of the `QUOTE` character to be parsed literally as part of a column's value. _`FORMAT CSV` only._
+`HEADER`  | `boolean`   | `boolean`  | Specifies that the file contains a header line with the names of each column in the file. On output, the first line contains the column names from the table, and on input, the first line is ignored.  _`FORMAT CSV` only._
```
Comment (philip-stoev) [inline doc/user/content/sql/copy-from.md:29]: due to https://github.com/MaterializeInc/materialize/issues/9278 we currently ignore the HEADER option on output. We also do not support FORMAT CSV for output. So, it is best to only mention input in here.

**Review by philip-stoev — CHANGES_REQUESTED** [review]
due to #9278 we currently ignore the HEADER option on output. We also do not support FORMAT CSV for output. So, it is best to only mention input in here.

File: doc/user/content/sql/copy-from.md (hunk 1 above)
Comment (msudberg) [reply doc/user/content/sql/copy-from.md]: @philip-stoev - Updated. Can you review latest commit?

**Review by philip-stoev — APPROVED** [review]
Thank you.

**Pull request merged**